crates/iceberg-catalog/src/service/storage/s3.rs

```diff
@@ -356,6 +423,82 @@ impl S3Profile {
         Ok(config)
     }
 
+    async fn get_aws_sts_token(
+        &self,
+        table_location: &str,
+        cred: &S3Credential,
+        arn: &String,
+    ) -> Result<aws_sdk_sts::types::Credentials, TableConfigError> {
+        self.get_sts_token(table_location, cred, Some(arn)).await
+    }
+
+    async fn get_minio_sts_token(
+        &self,
+        table_location: &str,
+        cred: &S3Credential,
+    ) -> Result<aws_sdk_sts::types::Credentials, TableConfigError> {
+        self.get_sts_token(table_location, cred, None).await
+    }
+
+    async fn get_sts_token(
+        &self,
+        table_location: &str,
+        cred: &S3Credential,
+        arn: Option<&str>,
+    ) -> Result<aws_sdk_sts::types::Credentials, TableConfigError> {
+        let cred = self
+            .get_config(self.get_aws_sdk_credentials(Some(cred))?)
+            .await;
+
+        let assume_role_builder = aws_sdk_sts::Client::new(&cred)
+            .assume_role()
+            .role_session_name("iceberg")
```

Comment:
This seems to be a really powerful tool. With the session name, we are able to end a session!
It shouldn't be part of this PR, but this would be a very nice feature. The role session name has a max. of 64 chars. We thus cannot encode user & location information in its name, but we can track sessions in a separate table holding unique session name, table_id, user_id and end that session if the user looses access.

Maybe we can create a follow-up?

Reply:
https://github.com/hansetag/iceberg-catalog/issues/247